TeamService.BuildNumber incorrectly using long

From the [AzureDevOps](https://docs.microsoft.com/en-us/azure/devops/pipelines/build/variables?view=azure-devops&tabs=yaml) documentation:

> Build.BuildNumber => The **name** of the completed build. You can specify the build number format that generates this value in the pipeline options.

When a build number uses a custom format there is no guarantee that this variable will actually be a number. IE: When I use a the following build number format: `$(Year:yyyy).$(Month).$(DayOfMonth)$(Rev:.r)`, I would expect a version along the lines of: "2019.5.20.1". With the current implementation this results in the following error: "##[error]Resolving parameter 'BUILD_BUILDNUMBER' failed."

I looked in the TeamServices class and found:
`public long BuildNumber => Variable<long>("BUILD_BUILDNUMBER");`

When I manually call `Variable<String>("BUILD_BUILDNUMBER")`, I get: "2019.5.20.5" and the build succeeds.

I therefor suggest fixing this by changing the BuildNumber to:
`public string BuildNumber => Variable<string>("BUILD_BUILDNUMBER");`

--- a/source/Nuke.Common/BuildServers/TeamServices.cs
+++ b/source/Nuke.Common/BuildServers/TeamServices.cs
@@ -1,4 +1,4 @@
-// Copyright 2019 Maintainers of NUKE.
+﻿// Copyright 2019 Maintainers of NUKE.
 // Distributed under the MIT License.
 // https://github.com/nuke-build/nuke/blob/master/LICENSE
 
@@ -40,7 +40,7 @@ namespace Nuke.Common.BuildServers
         public string AgentWorkFolder => Variable("AGENT_WORKFOLDER");
         public string ArtifactStagingDirectory => Variable("BUILD_ARTIFACTSTAGINGDIRECTORY");
         public long BuildId => Variable<long>("BUILD_BUILDID");
-        public long BuildNumber => Variable<long>("BUILD_BUILDNUMBER");
+        public string BuildNumber => Variable<string>("BUILD_BUILDNUMBER");
         public string BuildUri => Variable("BUILD_BUILDURI");
         public string BinariesDirectory => Variable("BUILD_BINARIESDIRECTORY");
         public string DefinitionName => Variable("BUILD_DEFINITIONNAME");

